integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala

```diff
@@ -243,11 +243,26 @@ object IndexServer extends ServerInterface {
   def main(args: Array[String]): Unit = {
     if (serverIp.isEmpty) {
       throw new RuntimeException(s"Please set the server IP to use Index Cache Server")
-    } else if (!isExecutorLRUConfigured) {
-      throw new RuntimeException(s"Executor LRU cache size is not set. Please set using " +
-                                 s"${ CarbonCommonConstants.CARBON_MAX_EXECUTOR_LRU_CACHE_SIZE }")
     } else {
       createCarbonSession()
+      if (!isExecutorLRUConfigured) {
```

Comment:
This is the wrong place to set the executor memory based on the %!!!
If you set here then only the driver knows about the CARBON_MAX_EXECUTOR_LRU_CACHE_SIZE. Executor would still use the default value.

Please move this to CacheProvider class.

Reply:
Done